prow/plugins/requiresig/requiresig.go

```diff
@@ -0,0 +1,151 @@
+
+	needsSIGMessage = "There are no sig labels on this issue. Please add a sig label."
+	needsSIGDetails = `A sig label can be added by either:
+
+1. mentioning a sig: ` + "`@kubernetes/sig-<group-name>-<group-suffix>`" + `
+    e.g., ` + "`@kubernetes/sig-contributor-experience-<group-suffix>`" + ` to notify the contributor experience sig, OR
+
+2. specifying the label manually: ` + "`/sig <group-name>`" + `
+    e.g., ` + "`/sig scalability`" + ` to apply the ` + "`sig/scalability`" + ` label
+
+Note: Method 1 will trigger an email to the group. See the [group list](https://github.com/kubernetes/community/blob/master/sig-list.md).
+The ` + "`<group-suffix>`" + ` in method 1 has to be replaced with one of these: _**bugs, feature-requests, pr-reviews, test-failures, proposals**_`
+)
+
+type githubClient interface {
+	BotName() (string, error)
+	AddLabel(org, repo string, number int, label string) error
+	RemoveLabel(org, repo string, number int, label string) error
+	CreateComment(org, repo string, number int, content string) error
+}
+
+type commentPruner interface {
+	PruneComments(shouldPrune func(github.IssueComment) bool)
+}
+
+func init() {
+	plugins.RegisterIssueHandler(pluginName, handleIssue, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, _ []string) (*pluginhelp.PluginHelp, error) {
+	// Only the 'Description' and 'Config' fields are necessary because this plugin does not react
+	// to any commands.
+	return &pluginhelp.PluginHelp{
+			Description: fmt.Sprintf("When a new issue is opened the requiresig plugin adds the %q label and leaves a comment requesting that a SIG (Special Interest Group) label be added to the issue. SIG labels are labels that have one of the following prefixes: %q.\nOnce a SIG label has been added to an issue, this plugin removes the %q label and deletes the comment it made previously.", needsSigLabel, labelPrefixes, needsSigLabel),
+			Config: map[string]string{
+				"": fmt.Sprintf("The comment the plugin creates includes this link to a list of the existing groups: %s", config.RequireSIG.GroupListURL),
+			},
+		},
+		nil
+}
+
+func handleIssue(pc plugins.PluginClient, ie github.IssueEvent) error {
+	return handle(pc.Logger, pc.GitHubClient, pc.CommentPruner, &ie)
+}
+
+func isSigLabel(label string) bool {
+	for i := range labelPrefixes {
+		if strings.HasPrefix(label, labelPrefixes[i]) {
+			return true
+		}
+	}
+	return false
+}
+
+func hasSigLabel(labels []github.Label) bool {
+	for i := range labels {
+		if isSigLabel(labels[i].Name) {
+			return true
+		}
+	}
+	return false
+}
+
+// handle is the workhorse notifying issue owner to add a sig label if there is none
+// The algorithm:
+// (1) return if this is not an opened, labelled, or unlabelled event or if the issue is closed.
+// (2) find if the issue has a sig label
+// (3) find if the issue has a needs-sig label
+// (4) if the issue has both the sig and needs-sig labels, remove the needs-sig label and comment.
```

Comment:
Do we need to comment here?

Reply:
I meant "remove both the needs-sig label and the comment".  I'll reword this to be more clear.